fix: Fix formatting + expand log drain docs (#2538)
18F/cg-site: _docs/deployment/logs.md

## How to automatically copy your logs elsewhere

If you want to set up your own storage for your application logs, you can set up [a "log drain" service](https://docs.cloudfoundry.org/devguide/services/log-management.html#user-provided) that sends the logs to your preferred location.

Create the user provided service and point it toward the endpoint where you want to send your logs:

      cf create-user-provided-service my-log-drain -l syslog-tls://<your-log-drain-service-endpont>

Then, bind the service you created to the app that you want connect:

      cf bind-service my-app my-log-drain

> If your log storage system cannot receive logs via syslog or https, it may be possible to forward

Reviewer: ```suggestion
If your log storage system cannot receive logs via `syslog` or `https`, it may be possible to forward
```

> the logs to an agent running on cloud.gov, which then sends the logs to your logging system.

Reviewer: ```suggestion
the logs to an application running on cloud.gov, which then sends the logs to your logging system.
```

To accomplish this, the agent must receive logs over https (since cloud.gov does not currently

> support raw TCP applications). When you do this, you *must* not bind the log drain service to the
> agent. Doing so may cause a logging feedback loop, where shipping logs to the agent generates logs
> which are then sent to the agent, and so on. This can quickly overwhelm your agent, likely causing
> dropped logs and unnecessary resource consumption.

Reviewer: ```suggestion
support raw TCP applications). 

When using a log drain with a log shipper application, you **must not bind the log drain service to the
log shipper application itself**. Doing so may cause a logging feedback loop, where shipping logs to the application generates logs
which are then sent to the application, and so on. This loop can quickly overwhelm your application, likely causing
dropped logs and unnecessary resource consumption.
```

> An example might look like:

Reviewer: ```suggestion
An example manifest for a log shipper application might look like:
```

```yml
---
# manifest.yml
---
applications:
- name: log-forwarder
  route: ((forwarder-route))
  memory: 256M
  stack: cflinuxfs4
  buildpacks:
    - binary_buildpack
```

> ```

Reviewer: ```suggestion
And the commands to create the log drain and bind it to your main application (not your log shipper application) would be:

```shell
```

## How to automatically copy your logs elsewhere

If you want to set up your own storage for your application logs, you can set up [a "log drain" service](https://docs.cloudfoundry.org/devguide/services/log-management.html#user-provided) that sends the logs to your preferred location.

Create the user provided service and point it toward the endpoint where you want to send your logs:

>       cf create-user-provided-service my-log-drain -l syslog-tls://<your-log-drain-service-endpont>

Reviewer: ```suggestion
      cf create-user-provided-service my-log-drain \
        -l syslog-tls://<your-log-drain-service-endpont>
```
![Screenshot 2024-09-05 at 3 15 36 PM](https://github.com/user-attachments/assets/108fa184-2845-4777-b402-ac9d0185d488)

Breaking command into multiple lines, otherwise the command text extends past the edge of the formatted code box

Then, bind the service you created to the app that you want connect:

      cf bind-service my-app my-log-drain

If your log storage system cannot receive logs via syslog or https, it may be possible to forward
the logs to an application running on cloud.gov, which then sends the logs to your logging system.

> To accomplish this, the agent must receive logs over https (since cloud.gov does not currently

Reviewer: ```suggestion
**To accomplish this, the application must receive logs over `https`** (since cloud.gov does not currently
```

support raw TCP applications).

When using a log drain with a log shipper application, you **must not bind the log drain service to the
log shipper application itself**. Doing so may cause a logging feedback loop, where shipping logs to the application generates logs
which are then sent to the application, and so on. This loop can quickly overwhelm your application, likely causing
dropped logs and unnecessary resource consumption.

An example manifest for a log shipper application might look like:
```yml
---
# manifest.yml
---
applications:
- name: log-forwarder
  route: ((forwarder-route))
  memory: 256M
  stack: cflinuxfs4
  buildpacks:
    - binary_buildpack
```

And the commands to create the log drain and bind it to your main application (not your log shipper application) would be:

```shell
$ cf push --var forwarder-route=logs.agency.example.gov

> $ cf create-user-provided-service main-log-drain -l https://logs.agency.example.gov

Reviewer: ```suggestion
$ cf create-user-provided-service main-log-drain \
  -l https://logs.agency.example.gov
```
